bin/init_exercise.py

```diff
@@ -0,0 +1,411 @@
+    # exiting like this isn't great for library use, but at least it's a quick fail
+    sys.exit(1)
+
+# check version info
+if sys.version_info[0] != 3 or sys.version_info[1] < 5:
+    print("This script requires Python 3.5 or higher", file=sys.stderr)
+    # exiting like this isn't great for library use, but at least it's a quick fail
+    sys.exit(1)
+
+
+def output_of(cmd, check_returncode=True):
+    "Return the stdout of the given command"
+    sp = subprocess.run(shlex.split(cmd),
+                        stdout=subprocess.PIPE,
+                        universal_newlines=True)
+    if check_returncode:
+        sp.check_returncode()
+    return sp.stdout.strip()
+
+
+REPO_ROOT = output_of('git rev-parse --show-toplevel')
+EXERCISES = os.path.join(REPO_ROOT, 'exercises')
+ITEM_NAME_CHARS = {c for c in string.ascii_lowercase + string.digits + '_'}
+VERSION = "0.1.0"
+
+
+def to_item_name(description):
+    "Produce a valid rust item name from arbitrary inputs"
+    item = description.lower().replace(' ', '_')
+    item = [c for c in item if c in ITEM_NAME_CHARS]
+    while len(item) > 0 and item[0] in string.digits:
+        item = item[1:]
+    if len(item) == 0:
+        raise ValueError("Could not produce an item name from " + description)
+    return item
+
+
+def url_for(name, file):
+    return "https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/{name}/{file}".format(
+        name=name, file=file,
+    )
+
+
+def get_problem_specification(name):
+    """
+    Try to get problem specifications for the exercise of the given name.
+
+    If the problem specifications repo doesn't exist, returns None.
+    Otherwise, returns a dict, of which the values might be None or str.
+    """
+    try:
+        with urllib.request.urlopen(url_for(name, 'canonical-data.json')) as response:
+            return json.loads(response.read())
+    except (urllib.request.URLError, json.JSONDecodeError):
+        pass
+
+
+@contextmanager
+def inside(path):
+    cwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
+
+
+def make_exercise(name):
+    "Make a new exercise with the specified name"
+    with inside(EXERCISES):
+        if os.path.exists(name):
+            print("{} already exists; aborting".format(name), file=sys.stderr)
+            sys.exit(1)
+        subprocess.run(['cargo', 'new', name])
+    exercise_dir = os.path.join(EXERCISES, name)
+    # blank out the default lib.rs
+    with inside(exercise_dir):
+        with open(os.path.join('src', 'lib.rs'), 'w') as lib_rs:
+            lib_rs.write('')
+        os.mkdir('tests')
+        with inside('tests'):
+            with open('{}.rs'.format(name), 'w') as tests_rs:
+                print("use {}::*;".format(name), file=tests_rs)
+                print(file=tests_rs)
+
+    cd = get_problem_specification(name)
+    if cd is None:
+        print("No problem specification for {} found".format(name))

```

Comment:
~~someone running this might think "oh no! does that mean it failed and did nothing of use?!" so I would suggest to make it clear that it did in fact do something after not finding the specification.~~

Reply:
> ~~someone running this might think "oh no! does that mean it failed and did nothing of use?!" so I would suggest to make it clear that it did in fact do something after not finding the specification.~~

I revoke my objection. It is handled, but in the respective functions (make_new_exercise prints "Creating new exercise from scratch")